ABIEvent should use indexed components

### What happened?

Either the ABIEvent type definition seems wrong to me or I am misunderstanding how it should be used.

The current definition is:

```python
class ABIEvent(TypedDict):
    """
    TypedDict to represent the `ABI` for an event.
    """

    name: str
    """Event name identifier."""
    type: Literal["event"]
    """Event ABI type."""
    anonymous: NotRequired[bool]
    """If True, event is anonymous. Cannot filter the event by name."""
    inputs: NotRequired[Sequence["ABIComponent"]]
    """Input components for the event."""```

But this is skipping the `indexed` argument of all event inputs.

Shouldn't it be

```diff
-    inputs: NotRequired[Sequence["ABIComponent"]]
+    inputs: NotRequired[Sequence["ABIComponentIndexed"]]
```

instead?

### Code that produced the error

```python
GRAPH_DELEGATION_TRANSFER_ABI: Final[Sequence[ABIEvent]] = [
    {  # all type ignores are due to inability of typing to see ABIComponentIndexed
        'anonymous': False,
        'inputs': [
            {
                'indexed': True,  # type: ignore
                'name': 'delegator',
                'type': 'address',
            },
            {
                'indexed': True,  # type: ignore
                'name': 'l2Delegator',
                'type': 'address',
            },
            {
                'indexed': True,  # type: ignore
                'name': 'indexer',
                'type': 'address',
            },
            {
                'indexed': False,  # type: ignore
                'name': 'l2Indexer',
                'type': 'address',
            },
            {
                'indexed': False,  # type: ignore
                'name': 'transferredDelegationTokens',
                'type': 'uint256',
            },
        ],
        'name': 'DelegationTransferredToL2',
        'type': 'event',
    },
]
```


### Full error output

```shell
rotkehlchen/chain/evm/decoding/thegraph/constants.py:22: error: Extra key "indexed" for TypedDict "ABIComponent"  [typeddict-unknown-key]
rotkehlchen/chain/evm/decoding/thegraph/constants.py:27: error: Extra key "indexed" for TypedDict "ABIComponent"  [typeddict-unknown-key]
rotkehlchen/chain/evm/decoding/thegraph/constants.py:32: error: Extra key "indexed" for TypedDict "ABIComponent"  [typeddict-unknown-key]
rotkehlchen/chain/evm/decoding/thegraph/constants.py:37: error: Extra key "indexed" for TypedDict "ABIComponent"  [typeddict-unknown-key]
```


### Fill this section in if you know how this could or should be fixed

```diff
-    inputs: NotRequired[Sequence["ABIComponent"]]
+    inputs: NotRequired[Sequence["ABIComponentIndexed"]]
```

### eth-typing Version

5.0.0

### Python Version

3.11.0

### Operating System

linux

### Output from `pip freeze`

```shell
not relevant
```

--- a/eth_typing/abi.py
+++ b/eth_typing/abi.py
@@ -58,7 +58,7 @@ class ABIEvent(TypedDict):
     """Event ABI type."""
     anonymous: NotRequired[bool]
     """If True, event is anonymous. Cannot filter the event by name."""
-    inputs: NotRequired[Sequence["ABIComponent"]]
+    inputs: NotRequired[Sequence["ABIComponentIndexed"]]
     """Input components for the event."""
 
 

--- a/newsfragments/92.bugfix.rst
+++ b/newsfragments/92.bugfix.rst
@@ -1,0 +1,1 @@
+``ABIEvent`` should use ``ABIComponentIndexed`` instead of ``ABIComponent`` as the type for ``inputs``.

